### PR TITLE
Add Code of Conduct

### PR DESCRIPTION
**Pull request by Gabriel439**
This is something I've been meaning to propose for a while but kept forgetting.

There's no particular reason I chose to add this now (i.e. this is not in response to
any conflict that I'm aware of).  If anything, it's probably better to propose this during a
quiet period so that the choice of whether to add a code of conduct doesn't become a
referendum on a particular dispute.

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+when an individual is representing the project or its community. Examples of
+representing a project or community include using an official project e-mail
+address, posting via an official social media account, or acting as an appointed
+representative at an online or offline event. Representation of a project may be
+further defined and clarified by project maintainers.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported by contacting the project team at Gabriel439@gmail.com. All
```
Comment (sjakobi) [inline CODE_OF_CONDUCT.md]: Are you the "project team", @Gabriel439?

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# Contributor Covenant Code of Conduct
```
Comment (sjakobi) [inline CODE_OF_CONDUCT.md]: I think this would be clearer:

```suggestion
# Code of Conduct
```

**Comment by Gabriel439** [comment]
@sjakobi: This only applies to this repository.  I eventually want to propose a code of conduct for `dhall-lang`, but I figured it would be better to test drive the code of conduct on our own project first

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (Gabriel439) [reply CODE_OF_CONDUCT.md]: Whoops!  It's a template that GitHub fills out for you after prompting for your e-mail.  I can either use a different e-mail or change the surrounding text

Comment (sjakobi) [reply CODE_OF_CONDUCT.md]: It's just a bit surprising to refer to a single person as a team I think. How about

```suggestion
reported by contacting the project maintainer at Gabriel439@gmail.com. All
```

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+address, posting via an official social media account, or acting as an appointed
+representative at an online or offline event. Representation of a project may be
+further defined and clarified by project maintainers.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported by contacting the project maintainer at Gabriel439@gmail.com. All
+complaints will be reviewed and investigated and will result in a response that
+is deemed necessary and appropriate to the circumstances. The project team is
```
Comment (sjakobi) [inline CODE_OF_CONDUCT.md]: The "project team" appears here again.

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+obligated to maintain confidentiality with regard to the reporter of an incident.
+Further details of specific enforcement policies may be posted separately.
+
+Project maintainers who do not follow or enforce the Code of Conduct in good
+faith may face temporary or permanent repercussions as determined by other
+members of the project's leadership.
```
Comment (sjakobi) [inline CODE_OF_CONDUCT.md]: This paragraph doesn't make a lot of sense when there's only a single maintainer.

**Comment by Gabriel439** [comment]
@sjakobi: I mean, there are multiple maintainers (at least you and I).  Maybe I can list both of our e-mails instead of just mine so somebody can contact either one of us to report CoC violations

**Comment by sjakobi** [comment]
> I mean, there are multiple maintainers (at least you and I). Maybe I can list both of our e-mails instead of just mine so somebody can contact either one of us to report CoC violations

Oh, I hadn't realized that I've become a maintainer! :)

> Project maintainers have the right and responsibility to remove, edit, or
reject comments, commits, code, wiki edits, issues, and other contributions
that are not aligned to this Code of Conduct, or to ban temporarily or
permanently any contributor for other behaviors that they deem inappropriate,
threatening, offensive, or harmful.

If you're comfortable with giving me the necessary permissions for the above actions, you can add my email address: simon.jakobi@gmail.com

In case I'm contacted as a member of the project team, I will always try to coordinate the response with you.

**Comment by Gabriel439** [comment]
@sjakobi: You already have those permissions.  You've been an admin on this project for a while (possibly unbeknownst to you) 🙂

**Comment by sjakobi** [comment]
Oh! Nice! I just really had never noticed! :)

**Pull request merged**